daml-foundations/daml-ghc/tests/ContractKeyTests.daml

```diff
@@ -0,0 +1,83 @@
+-- @IGNORE-LF 1.2
+daml 1.2
+module ContractKeyTests where
+
+-- (1) Breathing test.
+
+data Course =
+  Course with
+      institution : Party
+      title : Text
+  deriving (Show, Eq)
+
+data Registration =
+  Registration with
+      student : Party
+      course : Course
+      year : Int
+  deriving (Show, Eq)
+
+template Enrollment
+  with
+      reg : Registration
+  where
+      signatory reg.student, reg.course.institution
+      key reg : Registration
+      maintainer reg.course.institution
+
+-- (2) If `key` mentions `this.x` then `maintainer` can mention
+-- `this.x.y`.
+
+data M = M { m : Party } deriving (Show, Eq)
+data K = K { s : M } deriving (Show, Eq)
+
+template T
+  with
+    s, m : Party
+    k : K
+  where
+    signatory s
+    key k : K
+    maintainer k.s.m
+
+-- (3) Port './ContractKeys.daml' to DAML syntax.
```

Comment:
Not a useful comment - since no one will know what ContractKeys.daml is

Reply:
Ok